lib/utils.coffee

```diff
@@ -305,7 +294,7 @@ exports.requestAsync = (options, retriesRemaining = undefined) ->
 	requestTime = new Date()
 	p = fetch(url, opts)
 	if timeout
-		p = timeoutPromise(timeout, p)
+		p = p.timeout(timeout)
```

Comment:
This will throw a `Promise.TimeoutError` with the message `operation timed out`, so anyone checking for an error with the message `timeout` will now be broken - I think the new method is strictly superior but it would be worth at least noting it in the changelog